In [1]:
import os, ssl
if (not os.environ.get('PYTHONHTTPSVERIFY', '') and
    getattr(ssl, '_create_unverified_context', None)): 
    ssl._create_default_https_context = ssl._create_unverified_context
    
from urllib.request import urlopen
from bs4 import BeautifulSoup

url = 'http://www.innogov.go.kr/ucms/ogp/sug/list.do?pnum=1&menuNo=300011'
html = urlopen(url)
soup = BeautifulSoup(html, 'html.parser')

# result = soup.select_one("#content > div.suggestion_list > ul > li:nth-child(1) > dl > dt > p > a")
# print(result)
# print("제목 :", result.string)
# print("링크 :", result.attrs['href'])
results = soup.select("#content > div.suggestion_list > ul > li > dl > dt > p > a")
for i in results:
    print("제목 :", i.string)
    print("링크 :", i.attrs['href'])
    print()

제목 : 행정기관 소속 위원회 위원 개인정보 및 안건 공개 기준...
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=280&pnum=1

제목 : 초과근무 줄이기!
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=279&pnum=1

제목 : 원스톱 의원요구자료 제출 시스템 구축
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=278&pnum=1

제목 : 예산을 공평하게 분배
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=277&pnum=1

제목 : 공무원 인트라넷 통합메신저
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=276&pnum=1



In [2]:
from urllib.request import urlopen
from bs4 import BeautifulSoup
p = int(input("몇"))
for j in range(1, p+1):
    print(f"--------------------------{j}번째 페이지 --------------------------------")
    url = f'http://www.innogov.go.kr/ucms/ogp/sug/list.do?pnum={j}&menuNo=300011'
    html = urlopen(url)
    soup = BeautifulSoup(html, 'html.parser')

    results = soup.select("#content > div.suggestion_list > ul > li > dl > dt > p > a")
    for i in results:
        print("제목 :", i.string)
        print("링크 :", i.attrs['href'])
        print()

몇3
--------------------------1번째 페이지 --------------------------------
제목 : 행정기관 소속 위원회 위원 개인정보 및 안건 공개 기준...
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=280&pnum=1

제목 : 초과근무 줄이기!
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=279&pnum=1

제목 : 원스톱 의원요구자료 제출 시스템 구축
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=278&pnum=1

제목 : 예산을 공평하게 분배
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=277&pnum=1

제목 : 공무원 인트라넷 통합메신저
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=276&pnum=1

--------------------------2번째 페이지 --------------------------------
제목 : 임대아파트 이웃 주민들에게 피해를 주는 입주민에 대한 ...
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=275&pnum=2

제목 : 천차만별인 지역 유류가격을 지자체별로 저가 순위 주유소...
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=274&pnum=2

제목 : 행안부 생활공감정책추진단 제안에 대한 검토방식 개선
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=272&pnum=2

제목 : 주거형 오피스텔 차별 해소 필요
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=271&pnum=2

제목 : 국민연금불입금담보대출
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=270&pnum=2

--------------

In [3]:
from urllib.request import urlopen
from bs4 import BeautifulSoup
from urllib.parse import quote_plus

p = int(input("페이지: "))
search = input('검색어를 입력하세요: ')
for j in range(1, p+1):
    print(f"--------------------------{j}번째 페이지 --------------------------------")
    url = f'https://www.innogov.go.kr/ucms/ogp/sug/list.do?pnum={j}&menuNo=300011&cateCd=&status1Cd=&Status2Cd=&searchText={quote_plus(search)}&sugMonthTp=&orderKey=registDtDesc'
    html = urlopen(url)
    soup = BeautifulSoup(html, 'html.parser')

    results = soup.select("#content > div.suggestion_list > ul > li > dl > dt > p > a")
    for i in results:
        print("제목 :", i.string)
        print("링크 :", i.attrs['href'])
        print()

페이지: 10
검색어를 입력하세요: 개학
--------------------------1번째 페이지 --------------------------------
제목 : 교육기관의 업무 개시일을 1월 1일로 변경하여야 합니다...
링크 : /ucms/ogp/sug/view.do?menuNo=300011&sgId=164&pnum=1

--------------------------2번째 페이지 --------------------------------
--------------------------3번째 페이지 --------------------------------
--------------------------4번째 페이지 --------------------------------
--------------------------5번째 페이지 --------------------------------
--------------------------6번째 페이지 --------------------------------
--------------------------7번째 페이지 --------------------------------
--------------------------8번째 페이지 --------------------------------
--------------------------9번째 페이지 --------------------------------
--------------------------10번째 페이지 --------------------------------


In [10]:
from urllib.request import urlopen
from bs4 import BeautifulSoup
p = int(input("몇"))
link = []
for j in range(1, p+1):
    url = f'http://www.innogov.go.kr/ucms/ogp/sug/list.do?pnum={j}&menuNo=300011'
    html = urlopen(url)
    soup = BeautifulSoup(html, 'html.parser')

    results = soup.select("#content > div.suggestion_list > ul > li > dl > dt > p > a")
    for i in results:
        link.append(i.attrs['href'])
        
print(len(link))
link

몇10
50


['/ucms/ogp/sug/view.do?menuNo=300011&sgId=280&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=279&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=278&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=277&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=276&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=275&pnum=2',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=274&pnum=2',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=272&pnum=2',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=271&pnum=2',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=270&pnum=2',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=269&pnum=3',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=268&pnum=3',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=267&pnum=3',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=266&pnum=3',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=265&pnum=3',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=264&pnum=4',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=263&pnum=4',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=262&p

In [11]:
link[0:5]

['/ucms/ogp/sug/view.do?menuNo=300011&sgId=280&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=279&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=278&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=277&pnum=1',
 '/ucms/ogp/sug/view.do?menuNo=300011&sgId=276&pnum=1']

url = '/ucms/ogp/sug/view.do?menuNo=300011&sgId=277&pnum=1'
baseurl = f'https://www.innogov.go.kr/{url}'
print(baseurl)
print(baseurl)
html = urlopen(baseurl)
soup = BeautifulSoup(html, 'html.parser')
results = soup.select_one("#counter").string
title = soup.find('dl', class_='vveiw_titl1').find('dt')
print(title)
print(results)

url.split('=')[2].split('&')[0]

In [12]:
def find(url):
    article=[]
    baseurl = f'https://www.innogov.go.kr/{url}'
    html = urlopen(baseurl)
    soup = BeautifulSoup(html, 'html.parser')
    article.append(url.split('=')[2].split('&')[0]) # pnum
    article.append(soup.select_one('#content > div.vveiw_box1 > dl > dt').string) # 타이틀
    article.append(soup.select_one('#content > div.vveiw_box1 > div.vveiw_name > ul > li:nth-child(1) > dl > dd').string) #제안 분야
    article.append(soup.select_one('#content > div.vveiw_box1 > div.vveiw_name > ul > li:nth-child(2) > dl > dd').string) # 제안 시작
    article.append(soup.select_one('#content > div.vveiw_box1 > div.vveiw_name > ul > li:nth-child(3) > dl > dd').string) # 제안 마감
    article.append(soup.select_one("#counter").string ) # 추천 인원
    article.append(soup.select_one('#content > div.vveiw_box1 > div.vveiw_name > ul > li:nth-child(4) > dl > dd').string) # 제안인
    return article
    

data = []
for url in link:
    a = find(url)
    data.append(a)
    
print(data)

[['280', '행정기관 소속 위원회 위원 개인정보 및 안건 공개 기준에 관한 법적 근거 마련', '일반행정', '2019.10.16', '2020.01.13', '0', '법제혁신'], ['279', '초과근무 줄이기!', '일반행정', '2019.10.16', '2020.01.13', '2', 'ㅁㄴㅁ'], ['278', '원스톱 의원요구자료 제출 시스템 구축', '일반행정', '2019.10.15', '2020.01.12', '0', '연시랑'], ['277', '예산을 공평하게 분배', '사회통합', '2019.10.15', '2020.01.12', '3', '슈퍼맨119'], ['276', '공무원 인트라넷 통합메신저', '일반행정', '2019.10.15', '2020.01.12', '0', '페이소스'], ['275', '임대아파트 이웃 주민들에게 피해를 주는 입주민에 대한 처벌 강화를 요구합니다.', '공동체', '2019.10.07', '2020.01.04', '6', '범이찡'], ['274', '천차만별인 지역 유류가격을 지자체별로 저가 순위 주유소와 고가주유소를 공개하여 시민에게 저렴한 주유소를 알려 서민경제 부담을 줄여 주세요.', '공동체', '2019.10.03', '2019.12.31', '2', '청류'], ['272', '행안부 생활공감정책추진단 제안에 대한 검토방식 개선', '안전', '2019.09.20', '2019.12.18', '9', '주주단'], ['271', '주거형 오피스텔 차별 해소 필요', '일반행정', '2019.09.18', '2019.12.16', '4', 'vision'], ['270', '국민연금불입금담보대출', '보건복지', '2019.09.17', '2019.12.15', '6', '인생2모작'], ['269', '한 말씀', '상생협력', '2019.09.13', '2019.12.11', '2', '진상호'], ['268', '65세 경로대상자   사전안내', '보건복지', '2019.09.1

In [13]:
import pandas as pd
import numpy as np

column_list = ['id', 'title', 'category', 'start', 'end', 'vote', 'author']
df = pd.DataFrame(data, columns = column_list)
df.head()

id                                        title category       start  \
0  280  행정기관 소속 위원회 위원 개인정보 및 안건 공개 기준에 관한 법적 근거 마련     일반행정  2019.10.16   
1  279                                    초과근무 줄이기!     일반행정  2019.10.16   
2  278                         원스톱 의원요구자료 제출 시스템 구축     일반행정  2019.10.15   
3  277                                  예산을 공평하게 분배     사회통합  2019.10.15   
4  276                               공무원 인트라넷 통합메신저     일반행정  2019.10.15   

          end vote  author  
0  2020.01.13    0    법제혁신  
1  2020.01.13    2     ㅁㄴㅁ  
2  2020.01.12    0     연시랑  
3  2020.01.12    3  슈퍼맨119  
4  2020.01.12    0    페이소스

In [16]:
df.to_csv('../data/suggestion.csv', encoding='utf-8', sep=',')

In [17]:
data = pd.read_csv('../data/suggestion.csv', encoding = 'utf-8')
data

Unnamed: 0   id                                              title  \
0            0  280        행정기관 소속 위원회 위원 개인정보 및 안건 공개 기준에 관한 법적 근거 마련   
1            1  279                                          초과근무 줄이기!   
2            2  278                               원스톱 의원요구자료 제출 시스템 구축   
3            3  277                                        예산을 공평하게 분배   
4            4  276                                     공무원 인트라넷 통합메신저   
5            5  275        임대아파트 이웃 주민들에게 피해를 주는 입주민에 대한 처벌 강화를 요구합니다.   
6            6  274  천차만별인 지역 유류가격을 지자체별로 저가 순위 주유소와 고가주유소를 공개하여 시민...   
7            7  272                       행안부 생활공감정책추진단 제안에 대한 검토방식 개선   
8            8  271                                  주거형 오피스텔 차별 해소 필요   
9            9  270                                        국민연금불입금담보대출   
10          10  269                                               한 말씀   
11          11  268                                   65세 경로대상자   사전안내   
12          12  267                         한표.두표  10인 이내 투표 활용도   궁금!   
13          13  266                                 우리가  이용하는 식당 이상한점?   
14          14  265                                         빌고빌고 또 합니다   
15          15  264                                       암행국(어)사제도 신설   
16          16  263                           우리의  딩동.디디.알람  어디가 옳은가요?   
17          17  262                             주거지 자치제 시민 제안  개선점 의문점   
18          18  261                                      일용직   주거 실태파악   
19          19  260                     시간선택제공무원 근무시간 확대와 관련된 규정 개정 요청   
20          20  259                                 부동산 관리과 신설을 제안합니다.   
21          21  258                         국민 생각함    남여  독서실 화면창  개선안   
22          22  257                                           동사무소 번호표   
23          23  256                                       국민 생각함   자지체   
24          24  255                              운송업 고령자 운전면허 유지 조건 강화   
25          25  254               한컴오피스 없으면 대한민국 정부랑 일할수 없는 규제좀 풀어주세요.   
26          26  253                                        신호등 시간에 대하여   
27          27  252                                    게임 내 혐오표현 강화 처벌   
28          28  251                                              가짜 뉴스   
29          29  250                                     고령 운전자 면허 능력검사   
30          30  249                                너.나.내  .우리  파  벌  족   
31          31  248                                     중국 일명  조선족  제안   
32          32  247                                       주.야간 전투기 비행기   
33          33  246                                             국민의 판단   
34          34  245                                       기독교 목사 간음 행위   
35          35  244  부동산 확정일자, 민원24시(http://www.minwon.go.kr)에서 가능하...   
36          36  242                                       노인보행 안전대책 제안   
37          37  241             고령운전자문제 나이가아닌 의료적판단에따른 제한이 필요하다고 생각합니다   
38          38  240                                         청와대공무직공개채용   
39          39  239                          쓰레기통이   좀  많아졌으면   좋겠읍니다,   
40          40  238                       각종 대책에서 불구 하고 점점 떨어지고 있는 출산률   
41          41  237                                      여성 1인가구 안전 강화   
42          42  236               고령 운전자들의 운전안전 확보를 위해 재교육/평가를 실시해주세요.   
43          43  235                                             국민 생각함   
44          44  234                                        가로등 통반리  제안   
45          45  233                               외국인 등록증의 영어 이름을 바꿔요!   
46          46  232                             정부1번가 /시민 제안  차이점 있는지,   
47          47  231                                          횡단 보도 정지선   
48          48  230               교차로 신호등 점멸시 교차로 차량통과 시속30km이하로 제한해요.   
49          49  229                        유기동물 안락사 문제를 정부 차원에서 해결해주세요   

   category       start         end  vote      author  
0      일반행정  2019.10.16  2020.01.13     0        법제혁신  
1      일반행정  2019.10.16  2020.01.13     2         ㅁㄴㅁ  
2      일반행정  2019.10.15  2020.01.12     0         연시랑  
3    

In [18]:
df.rename(columns={df.columns[1]:'제목', df.columns[2]:'제안분야',
                  df.columns[3]:'제안시작', df.columns[4]:'제안마감',
                  df.columns[5]:'투표수', df.columns[6]:'제안인'}, inplace=True
                  )
df.head()

id                                           제목  제안분야        제안시작  \
0  280  행정기관 소속 위원회 위원 개인정보 및 안건 공개 기준에 관한 법적 근거 마련  일반행정  2019.10.16   
1  279                                    초과근무 줄이기!  일반행정  2019.10.16   
2  278                         원스톱 의원요구자료 제출 시스템 구축  일반행정  2019.10.15   
3  277                                  예산을 공평하게 분배  사회통합  2019.10.15   
4  276                               공무원 인트라넷 통합메신저  일반행정  2019.10.15   

         제안마감 투표수     제안인  
0  2020.01.13   0    법제혁신  
1  2020.01.13   2     ㅁㄴㅁ  
2  2020.01.12   0     연시랑  
3  2020.01.12   3  슈퍼맨119  
4  2020.01.12   0    페이소스

In [19]:
data = pd.read_csv('../data/suggestion.csv', encoding = 'utf-8')
data

Unnamed: 0   id                                              title  \
0            0  280        행정기관 소속 위원회 위원 개인정보 및 안건 공개 기준에 관한 법적 근거 마련   
1            1  279                                          초과근무 줄이기!   
2            2  278                               원스톱 의원요구자료 제출 시스템 구축   
3            3  277                                        예산을 공평하게 분배   
4            4  276                                     공무원 인트라넷 통합메신저   
5            5  275        임대아파트 이웃 주민들에게 피해를 주는 입주민에 대한 처벌 강화를 요구합니다.   
6            6  274  천차만별인 지역 유류가격을 지자체별로 저가 순위 주유소와 고가주유소를 공개하여 시민...   
7            7  272                       행안부 생활공감정책추진단 제안에 대한 검토방식 개선   
8            8  271                                  주거형 오피스텔 차별 해소 필요   
9            9  270                                        국민연금불입금담보대출   
10          10  269                                               한 말씀   
11          11  268                                   65세 경로대상자   사전안내   
12          12  267                         한표.두표  10인 이내 투표 활용도   궁금!   
13          13  266                                 우리가  이용하는 식당 이상한점?   
14          14  265                                         빌고빌고 또 합니다   
15          15  264                                       암행국(어)사제도 신설   
16          16  263                           우리의  딩동.디디.알람  어디가 옳은가요?   
17          17  262                             주거지 자치제 시민 제안  개선점 의문점   
18          18  261                                      일용직   주거 실태파악   
19          19  260                     시간선택제공무원 근무시간 확대와 관련된 규정 개정 요청   
20          20  259                                 부동산 관리과 신설을 제안합니다.   
21          21  258                         국민 생각함    남여  독서실 화면창  개선안   
22          22  257                                           동사무소 번호표   
23          23  256                                       국민 생각함   자지체   
24          24  255                              운송업 고령자 운전면허 유지 조건 강화   
25          25  254               한컴오피스 없으면 대한민국 정부랑 일할수 없는 규제좀 풀어주세요.   
26          26  253                                        신호등 시간에 대하여   
27          27  252                                    게임 내 혐오표현 강화 처벌   
28          28  251                                              가짜 뉴스   
29          29  250                                     고령 운전자 면허 능력검사   
30          30  249                                너.나.내  .우리  파  벌  족   
31          31  248                                     중국 일명  조선족  제안   
32          32  247                                       주.야간 전투기 비행기   
33          33  246                                             국민의 판단   
34          34  245                                       기독교 목사 간음 행위   
35          35  244  부동산 확정일자, 민원24시(http://www.minwon.go.kr)에서 가능하...   
36          36  242                                       노인보행 안전대책 제안   
37          37  241             고령운전자문제 나이가아닌 의료적판단에따른 제한이 필요하다고 생각합니다   
38          38  240                                         청와대공무직공개채용   
39          39  239                          쓰레기통이   좀  많아졌으면   좋겠읍니다,   
40          40  238                       각종 대책에서 불구 하고 점점 떨어지고 있는 출산률   
41          41  237                                      여성 1인가구 안전 강화   
42          42  236               고령 운전자들의 운전안전 확보를 위해 재교육/평가를 실시해주세요.   
43          43  235                                             국민 생각함   
44          44  234                                        가로등 통반리  제안   
45          45  233                               외국인 등록증의 영어 이름을 바꿔요!   
46          46  232                             정부1번가 /시민 제안  차이점 있는지,   
47          47  231                                          횡단 보도 정지선   
48          48  230               교차로 신호등 점멸시 교차로 차량통과 시속30km이하로 제한해요.   
49          49  229                        유기동물 안락사 문제를 정부 차원에서 해결해주세요   

   category       start         end  vote      author  
0      일반행정  2019.10.16  2020.01.13     0        법제혁신  
1      일반행정  2019.10.16  2020.01.13     2         ㅁㄴㅁ  
2      일반행정  2019.10.15  2020.01.12     0         연시랑  
3    